test/test_datasets.py

```diff
@@ -141,76 +145,68 @@ def test_places365_repr_smoke(self):
             self.assertIsInstance(repr(dataset), str)
 
 
-class STL10Tester(DatasetTestcase):
-    @contextlib.contextmanager
-    def mocked_root(self):
-        with stl10_root() as (root, data):
-            yield root, data
+class STL10Tester(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.STL10
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        split=("train", "test", "unlabeled", "train+unlabeled"))
 
-    @contextlib.contextmanager
-    def mocked_dataset(self, pre_extract=False, download=True, **kwargs):
-        with self.mocked_root() as (root, data):
-            if pre_extract:
-                utils.extract_archive(os.path.join(root, data["archive"]))
-            dataset = torchvision.datasets.STL10(root, download=download, **kwargs)
-            yield dataset, data
-
-    def test_not_found(self):
-        with self.assertRaises(RuntimeError):
-            with self.mocked_dataset(download=False):
-                pass
-
-    def test_splits(self):
-        for split in ('train', 'train+unlabeled', 'unlabeled', 'test'):
-            with self.mocked_dataset(split=split) as (dataset, data):
-                num_images = sum([data["num_images_in_split"][part] for part in split.split("+")])
-                self.generic_classification_dataset_test(dataset, num_images=num_images)
-
-    def test_folds(self):
-        for fold in range(10):
-            with self.mocked_dataset(split="train", folds=fold) as (dataset, data):
-                num_images = data["num_images_in_folds"][fold]
-                self.assertEqual(len(dataset), num_images)
-
-    def test_invalid_folds1(self):
-        with self.assertRaises(ValueError):
-            with self.mocked_dataset(folds=10):
-                pass
-
-    def test_invalid_folds2(self):
-        with self.assertRaises(ValueError):
-            with self.mocked_dataset(folds="0"):
-                pass
-
-    def test_transforms(self):
-        expected_image = "image"
-        expected_target = "target"
+    def inject_fake_data(self, tmpdir, config):
+        def make_binary_file(num_elements, root, name):
```

Comment:
Is there a reason to define these functions locally instead as instance methods? If not, I would prefer the latter for readability.

Reply:
I think the main reason is that these functions are only ever used in `inject_fake_data`. Making them a method (even private) would hint that these may be used in more than one place. Also, they don't use or mutate `self` at all.

I tend to favor nested functions for this as well, although I noticed you had a different opinion already :p

These are just tests so no strong opinion from me anyway

Reply:
I wouldn't be against it if it is only a small function. But in this case these functions make up most of the body of `inject_fake_data`. I actually had to scroll down from the definition to reach the code that is actually executed. 

> Making them a method (even private) would hint that these may be used in more than one place.

Not necessarily. You can simply factor out code to enhance the readability. For example look at the `download` method. We only call this once during instantiation, but it is super easy to read

https://github.com/pytorch/vision/blob/eee424922e998dddcaabe927e823ec515aa2927b/torchvision/datasets/stl10.py#L62-L63

without having a massive constructor.

> Also, they don't use or mutate self at all.

I don't mind if they are static.